imsim/stamp.py

```diff
@@ -536,3 +584,181 @@ def draw(self, prof, image, method, offset, config, base, logger):
+                    "maxN": maxN,
+                    "rng": self.rng,
+                    "n_subsample": 1,
+                })
+
+            # Go back to a combined convolution for fft drawing.
+            prof = galsim.Convolve([obj] + psfs)
+            try:
+                fft_image = prof.drawImage(bandpass, **kwargs)
+            except galsim.errors.GalSimFFTSizeError as e:
+                # I think this shouldn't happen with the updates I made to how the image size
+                # is calculated, even for extremely bright things.  So it should be ok to
+                # just report what happened, give some extra information to diagonose the problem
+                # and raise the error.
+                logger.error('Caught error trying to draw using FFT:')
+                logger.error('%s',e)
+                logger.error('You may need to add a gsparams field with maximum_fft_size to')
+                logger.error('either the psf or gal field to allow larger FFTs.')
+                logger.info('prof = %r',prof)
+                logger.info('fft_image = %s',fft_image)
+                logger.info('offset = %r',offset)
+                raise
+            # Some pixels can end up negative from FFT numerics.  Just set them to 0.
+            fft_image.array[fft_image.array < 0] = 0.
+            if self.diffraction_fft:
+                self.diffraction_fft.apply(fft_image, bandpass.effective_wavelength)
+            fft_image.addNoise(galsim.PoissonNoise(rng=self.rng))
+            # In case we had to make a bigger image, just copy the part we need.
+            image += fft_image[image.bounds]
+            base['realized_flux'] = fft_image.added_flux
+        else:
+            obj = obj.withFlux(obj_info.phot_flux, initial_flux_bandpass)
+
+            if not faint and 'photon_ops' in config:
+                photon_ops = galsim.config.BuildPhotonOps(config, 'photon_ops', base, logger)
+            else:
+                photon_ops = []
+
+            if self.do_reweight:
+                photon_ops.append(
+                    BandpassRatio(
+                        target_bandpass=bandpass,
+                        initial_bandpass=self.fiducial_bandpass,
+                    )
+                )
+                bp_for_drawImage = self.fiducial_bandpass
+            else:
+                bp_for_drawImage = bandpass
+
+            photon_ops = psfs + photon_ops
+            
+            rng = galsim.config.GetRNG(config, base, logger, "LSST_Silicon")
+            obj.drawImage(bp_for_drawImage,
+                        method='phot',
+                        offset=offset,
+                        rng=rng,
+                        maxN=None,
+                        n_photons=obj_info.phot_flux,
+                        image=image,
+                        wcs=base['wcs'],
+                        sensor=NullSensor(), # Prevent premature photon accumulation
+                        photon_ops=psfs,
```

Comment:
Should this be
```
photon_ops=photon_ops,
```
?

Reply:
You're right, it should be `photon_ops`, but the code above which creates the initial `photon_ops` list from the config is incorrect -- thanks for noticing something's not right here. It should be contain the PSF and the BandpassRatio photon op (only if the SED needs to be reweighted), which need to be dealt with here. All the other photon ops are dealt with later in the pipeline and applied to the whole photon batch. I'm fixing this now.